example/multi-jvm-example/multi-jvm-example-docker-runner/src/test/java/uk/gov/gchq/palisade/example/MultiJvmDockerExampleIT.java

```diff
@@ -57,14 +96,46 @@ public void shouldReadAsAlice() throws Exception {
     }
 
     private ConfigurationService getConfigurationService() {
-        return ExampleConfigurator.setupMultiJVMConfigurationService(Collections.singletonList("http://localhost:2379"),
-                Optional.empty(),
-                Optional.of(new ProxyRestPolicyService("http://policy-service:8080/policy")),
-                Optional.of(new ProxyRestUserService("http://user-service:8080/user")),
-                Optional.of(new ProxyRestResourceService("http://resource-service:8080/resource")),
-                Optional.of(new ProxyRestPalisadeService("http://palisade-service:8080/palisade")),
-                Optional.of(new SimpleCacheService().backingStore(new EtcdBackingStore().connectionDetails(Collections.singletonList("http://etcd:2379"), false)))
-        );
+        EtcdBackingStore store = null;
+        try {
+            store = new EtcdBackingStore().connectionDetails(Collections.singletonList("http://localhost:2379"));
+            AuditService audit = new LoggerAuditService();
+            PolicyService policy = new ProxyRestPolicyService("http://localhost:8081/policy");
+            UserService user = new ProxyRestUserService("http://localhost:8083/user");
+            ResourceService resource = new ProxyRestResourceService("http://localhost:8082/resource");
+            PalisadeService palisade = new ProxyRestPalisadeService("http://localhost:8080/palisade");
+            CacheService cache = new SimpleCacheService().backingStore(store);
+            ConfigurationService config = new ProxyRestConfigService("http://localhost:8085/config");
+            final ServiceConfiguration sc = new ServiceConfiguration();
+
+            //each service to write their configuration into the initial configuration
+            Collection<Service> services = Stream.of(audit, user, resource, policy, palisade, cache, config).collect(Collectors.toList());
+            services.forEach(service -> service.recordCurrentConfigTo(sc));
+
+            CacheService dockerCacheService = new SimpleCacheService().backingStore(new EtcdBackingStore().connectionDetails(Collections.singleton("http://etcd:2379"), false));
+
+            ExampleConfigurator.setupMultiJVMConfigurationService(
+                    new ProxyRestPolicyService("http://policy-service:8080/policy"),
+                    new ProxyRestUserService("http://user-service:8080/user"),
+                    new ProxyRestResourceService("http://resource-service:8080/resource"),
+                    new ProxyRestPalisadeService("http://palisade-service:8080/palisade"),
+                    dockerCacheService,
+                    new ProxyRestConfigService("http://localhost:8085/config"),
```

Comment:
Should these either all be localhost addresses or internal docker addresses? Currently you have a mix which doesn't look right.